node/pkg/watchers/solana/client.go

```diff
@@ -645,9 +634,9 @@ func (s *SolanaWatcher) fetchBlock(ctx context.Context, logger *zap.Logger, slot
 }
 
 // processTransaction processes a transaction and publishes any Wormhole events.
-func (s *SolanaWatcher) processTransaction(ctx context.Context, logger *zap.Logger, tx *solana.Transaction, meta *rpc.TransactionMeta, slot uint64, isReobservation bool) {
+func (s *SolanaWatcher) processTransaction(ctx context.Context, logger *zap.Logger, rpcClient *rpc.Client, tx *solana.Transaction, meta *rpc.TransactionMeta, slot uint64, isReobservation bool) (numObservations uint32) {
```

Comment:
Would it make sense to access `s.rpcClient` here instead? (And `s.logger` now that it exists)

Reply:
Would work for `s.logger` but not for `s.rpcClient` because we need to use the custom one when we are reobserving.

Reply:
I confirmed that I do need to pass in `rpcClient`, but not `logger`. I made that change.